### PR TITLE
Improve animation support in fromLibGdx()

### PR DESCRIPTION
**Pull request by Descrout**
Get the image index in the atlas, if its not -1, this means the image is a part of an animation. Add `_$index` suffix to name so it can be used by `animation.addByPrefix`.

### Example

**Code:**
```haxe
var frames = FlxAtlasFrames.fromLibGdx(AssetPaths.myAtlas__png, AssetPaths.myAtlas__atlas);
var player = new FlxSprite();
player.frames = frames;
player.animation.addByPrefix("idle", "adventurer-idle", 12);
player.animation.addByPrefix("crouch", "adventurer-crouch", 12);
player.animation.play("idle");
```

**myAtlas.atlas :** 
```
myAtlas.png
size: 2048,512
format: RGBA8888
filter: Nearest,Nearest
repeat: none
adventurer-idle
  rotate: false
  xy: 324, 51
  size: 50, 37
  orig: 50, 37
  offset: 0, 0
  index: 2
adventurer-idle
  rotate: false
  xy: 568, 53
  size: 50, 37
  orig: 50, 37
  offset: 0, 0
  index: 0
adventurer-idle
  rotate: false
  xy: 1974, 391
  size: 50, 37
  orig: 50, 37
  offset: 0, 0
  index: 1
adventurer-crouch
  rotate: false
  xy: 1981, 430
  size: 50, 37
  orig: 50, 37
  offset: 0, 0
  index: 2
adventurer-crouch
  rotate: false
  xy: 1870, 391
  size: 50, 37
  orig: 50, 37
  offset: 0, 0
  index: 0
adventurer-crouch
  rotate: false
  xy: 1922, 391
  size: 50, 37
  orig: 50, 37
  offset: 0, 0
  index: 1
```

### CODE REVIEW

File: flixel/graphics/frames/FlxAtlasFrames.hx [hunk 1]
```diff
@@ -122,15 +122,14 @@ class FlxAtlasFrames extends FlxFramesCollection
...
+	 * @param   source        The image source (can be `FlxGraphic`, `String` or `BitmapData`).
+	 * @param   description   Contents of the file with atlas description.
 	 *                        You can get it with `Assets.getText(path/to/description/file)`.
 	 *                        Or you can just pass path to the description file in the assets directory.
 	 * @return  Newly created `FlxAtlasFrames` collection.
 	 */
-	public static function fromLibGdx(Source:FlxGraphicAsset, Description:String):FlxAtlasFrames
-	{
-		var graphic:FlxGraphic = FlxG.bitmap.add(Source);
+	public static function fromLibGdx(source:FlxGraphicAsset, description:String):FlxAtlasFrames {
```
Comment (Gama11) [inline flixel/graphics/frames/FlxAtlasFrames.hx:131]: Please follow the code style. You can just run haxe-formatter over it.

File: flixel/graphics/frames/FlxAtlasFrames.hx [hunk 2]
```diff
@@ -212,18 +208,17 @@ class FlxAtlasFrames extends FlxFramesCollection
 	 * Internal method for LibGDX atlas parsing. It tries to extract dimensions info from specified string.
 	 *
 	 * @param   line   `String` to extract info from.
-	 * @param   size   `Array` to store extracted info to.
 	 * @return  `Array` filled with dimensions info.
 	 */
-	static function getDimensions(line:String, size:Array<Int>):Array<Int>
+	static function getDimensions(line:String):Array<Int>
```
Comment (Gama11) [inline flixel/graphics/frames/FlxAtlasFrames.hx:213]: Might as well make this return a FlxPoint a this point?

Comment (Gama11) [reply flixel/graphics/frames/FlxAtlasFrames.hx:213]: Or `{x: Int, y:Int}` if it's important that they're ints.

Comment (Descrout) [reply flixel/graphics/frames/FlxAtlasFrames.hx:213]: Yes let's do that :+1:

File: flixel/graphics/frames/FlxAtlasFrames.hx [hunk 3]
```diff
@@ -208,7 +208,7 @@ class FlxAtlasFrames extends FlxFramesCollection
 	 * Internal method for LibGDX atlas parsing. It tries to extract dimensions info from specified string.
 	 *
 	 * @param   line   `String` to extract info from.
-	 * @return  `Array` filled with dimensions info.
+	 * @return  `{x:Int, y:Int}` dimensions info.
```
Comment (Gama11) [inline flixel/graphics/frames/FlxAtlasFrames.hx:211]: Actually, there shouldn't be a type here at all, this isn't JSDoc.

**Pull request merged**